sdk-core/message.md

```diff
@@ -3,14 +3,38 @@
 ```
 dto Message:
     data: bytes;
-    signature: bytes;
+    signature: Optional[bytes];
+    address: Optional[IAddress];
+    version: Optional[int]
 ```
 
+The only manadatory parameter for the constructor it's `data`. The others can be `undefined` or have default empty values.
+If `version` is not provided the default value should be `1`.
+
 ## MessageComputer
 
 ```
 class MessageComputer:
     compute_bytes_for_signing(message: Message): bytes;
+
+    // return the call of `compute_bytes_for_signing()`
+    compute_bytes_for_verifying(messge: Message): bytes;
```

Comment:
question for reviewers:
Should we maybe have just one method called `serialize()` or something similar instead of `compute_bytes_for_...()`?

Reply:
This would mean that also the name of the class should be changed, right?

I know that we have a message that has some serialize... methods in it. Will this replace that one?

Reply:
I didn't think about changing the name of the class, but now it makes sense to have methods called `compute...`, even though the underlying implementation is the same.

I assume that you are talking about [SignableMessage](https://github.com/multiversx/mx-sdk-js-core/blob/feat/next/src/signableMessage.ts), and yes, this class will replace that one, but in the future. For the moment we will have both.

Reply:
`serialize()` sounds good, yes. The name of the class should stay the same for symmetry / consistency with other `*Computer` classes, which sometimes do other things as well (e.g. compute shard of address, compute hash of transaction).

On the other hand, this is not an actual serialization, since it destroys the data irreversibly. Thus, the name is also good as it is now.

Reply:
will leave it as it is for now, then.